pipeline-marketplace-images/github-sign-off/create-issue

```diff
@@ -2,29 +2,56 @@
 
 set -euo pipefail
 
-if [ -f /kratix/metadata/auto-approve ]; then
-  echo "Auto-approve enabled. Skipping issue creation..."
-  exit 0
-fi
+function create_issue() {
+  if [ -f /kratix/metadata/skip-approval ]; then
+    echo "Skipping approval process as per configuration..."
+    return 2
+  fi
 
-currentRequest=$(yq -r '.status.approvalIssue' /kratix/input/object.yaml)
-if [ "$currentRequest" != "null" ]; then
-  echo "Approval issue already exists (or auto-approved). Skipping issue creation..."
-  exit 0
-fi
+  if [ -f /kratix/metadata/auto-approve ]; then
+    echo "Auto-approve enabled. Skipping issue creation..."
+    return 2
+  fi
+
+  currentRequest=$(yq -r '.status.approvalIssue' /kratix/input/object.yaml)
+  if [ "$currentRequest" != "null" ]; then
+    echo "Approval issue already exists (or auto-approved). Skipping issue creation..."
+    return 1
+  fi
+
+  {
+    echo ""
+    echo "Close the issue to accept; Close it as not-planned to reject"
+    echo ""
+    echo '```yaml'
+    yq 'del(.status)' /kratix/input/object.yaml
+    echo '```'
+  } > issue-body
+
+  issue=$(gh issue create \
+    --repo "$GITHUB_REPOSITORY" \
+    --title "New $(yq -r '.kind' /kratix/input/object.yaml) request waiting approval" \
+    --body-file issue-body)
 
-echo "New $(yq -r '.kind' /kratix/input/object.yaml) requested"
-echo "" >> issue-body
-echo "Close the issue to accept; Close it as not-planned to reject" >> issue-body
-echo "" >> issue-bod
-echo '```yaml' >> issue-body
-cat /kratix/input/object.yaml | yq 'del(.status)' >> issue-body
-echo '```' >> issue-body
-
-issue=$(gh issue create \
-  --repo $GITHUB_REPOSITORY \
-  --title "New $(yq -r '.kind' /kratix/input/object.yaml) request waiting approval" \
-  --body-file issue-body)
-
-echo "approvalIssue: $(echo $issue)" > /kratix/metadata/status.yaml
-echo "message: Waiting approval. Check $(echo $issue) for details" >> /kratix/metadata/status.yaml
+  echo "$issue"
+  return 0
```

Comment:
Should this return based on the issue create status?

Reply:
Not sure I follow what your asking 🤔  the return is the exit code, the issue is being returned by it being echo'd. Do you mean we should check something about the issue before returning it?